.pre-commit-config.yaml

```diff
@@ -14,16 +13,11 @@ repos:
       - id: mixed-line-ending
       - id: check-case-conflict
   - repo: https://github.com/psf/black
-    rev: 22.6.0
+    rev: 23.7.0
     hooks:
       - id: black
-  - repo: https://github.com/pycqa/isort
-    rev: 5.12.0
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.0.283
     hooks:
-      - id: isort
-        args: ['--profile', 'black']
-  - repo: https://github.com/pycqa/flake8
-    rev: 6.0.0
-    hooks:
-      - id: flake8
-        entry: poetry run flake8
+      - id: ruff
+        args: [ --fix, --exit-non-zero-on-fix ]
```

Comment:
For a pre-commit hook, does `--exit-non-zero-on-fix` make sense? If you want to be lazy and let the hook fix for you, it won't work as the non-zero exit will prevent the commit from being written?

Reply:
good question, i was going of some docs and it works for me as expected, just as black, it fixes it for you automatically

Reply:
i think it just to throw you back and you need to stage the files again and then commit so you see what was done before you actually commit it